daisy/workflow/step_create_disks.go

```diff
@@ -82,11 +86,21 @@ func (c *CreateDisks) run(w *Workflow) error {
 				name = w.genName(cd.Name)
 			}
 
+			zone := w.Zone
+			if cd.Zone != "" {
+				zone = cd.Zone
+			}
+
+			project := w.Project
+			if cd.Project != "" {
+				project = cd.Project
+			}
+
 			// Get the source image link.
 			var imageLink string
 			var image *resource
 			var err error
-			if cd.SourceImage == "" || isLink(cd.SourceImage) {
+			if cd.SourceImage == "" || (isLink(cd.SourceImage) && imageValid(w, cd.SourceImage)) {
```

Comment:
Is the "isLink" necessary here anymore since you are using URL regexes in the *Valid functions?